lib/actions/local/configure.coffee

```diff
@@ -95,100 +67,65 @@ inquirerOptions = (data) -> [
 	}
 ]
 
+joinPath = (path) -> '/' + path.join('/')
+
+load = (target, type) ->
+	{ Reconfix } = require('reconfix-preview')
+	imagefs = require('resin-image-fs')
+
+	reconfix = new Reconfix({
+		read: (partition, path) ->
+			imagefs.readFile
+				image: target
+				partition: partition
+				path: joinPath(path)
+			.catch (err) ->
+				if err?.code == 'NOENT'
+					return Buffer.alloc(0)
+				else
+					throw err
+			.then (content) -> Buffer.from(content, 'utf-8')
+			.catch (err) -> {}
+		write: (partition, path, data) ->
+			imagefs.writeFile
+				image: target,
+				partition: partition,
+				path: joinPath(path)
+			, data.toString('utf-8')
+	})
+	readFileAsync(path.join(__dirname, '../../schemas/resin-os.json'))
+	.then (content) -> JSON.parse(content)
+	.then (json) ->
+		reconfix.loadSchema(json)
+		return reconfix
+
+
+readConfiguration = (target, type) ->
+	load(target, type)
+	.then (reconfix) -> reconfix.readValues()
+
+writeConfiguration = (config, target, type) ->
+	load(target, type)
+	.then (reconfix) -> reconfix.writeValues(config)
+
+
 getConfiguration = (data) ->
-	_ = require('lodash')
 	inquirer = require('inquirer')
 
+	console.log('read configuration: ' + JSON.stringify(data, '  '))
+
 	# `persistentLogging` can be `undefined`, so we want
 	# to make sure that case defaults to `false`
 	data = _.assign data,
 		persistentLogging: data.persistentLogging or false
 
 	inquirer.prompt(inquirerOptions(data))
+	.then (answers) ->
+		keys = _.filter(_.keys(answers), (key) -> key != 'advancedSettings')
```

Comment:
I think this could be simplified to `_.keys(_.omit(answers, [ 'advancedSettings' ]))`